Adding up budget-not-provided separately #537

Ocre42 opened the pull request:
Addressing #537 
Forwardlooking activities with budget-not-provided will be added up separately instead of replacing the activities that provide budgets.

andylolz commented on forwardlooking.py:
```diff
@@ -33,34 +33,35 @@ def generate_row(publisher):
-                row['year_columns'][1][year] = by_hierarchy[hierarchies_with_nonzero_budgets[0]][forwardlooking_budget].get(year) or 0
+                row['year_columns'][1][year] = by_hierarchy[hierarchies_with_nonzero_budgets[0]]['forwardlooking_activities_with_budgets'].get(year) or 0
+                if row['budget_not_provided']:
+                    by_hierarchy[hierarchies_with_nonzero_budgets[0]]['forwardlooking_activities_with_budget_not_provided'].get(year) or 0
```
Something doesn’t look quite right here – there’s no equals sign on this line. Presumably this is supposed to say:

```python
row['year_columns'][1][year] = …
```

…or similar?

andylolz replied:
Looks like this has been addressed in #556.

andylolz commented:
I don’t think this PR attempts to address #537. I.e. it doesn’t attempt to split the figure for activities that include a budget from the figure for activities that explicitly declare `budget-not-provided`.

I think it might instead be addressing https://github.com/IATI/IATI-Dashboard/issues/537#issuecomment-477270154. Is that correct?

Ocre42 commented:
Yes, this addresses the comment specifically, but there is no intention to separate the `budget-not-provided` attribute from the `budgets` in calculations, as this is done to follow the existing methodology.

andylolz commented:
Okay, that’s fine. I guess #537 can be closed as a wontfix, then.
